microcosm_flask/sentry.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from os import environ
+from typing import Any, Optional
+from unittest.mock import MagicMock
+
+from microcosm.decorators import defaults
+from microcosm.object_graph import ObjectGraph
+from microcosm_logging.decorators import logger
+
+from sentry_sdk.utils import BadDsn
+
+
+@dataclass
+class SentryConfig:
+    dsn: Optional[str] = None
+    enabled: bool = False
+    client: Optional[Any] = None
+
+
+def before_send(event, hint):
+    """
+    Before sending a event to sentry scrub all values that are non-uuid data from the event
+    """
+    for exception in event["exception"]["values"]:
+        for frame in exception["stacktrace"]["frames"]:
+            for var_name, value in frame["vars"].items():
+                if var_name in ["self", "cls"]:
+                    continue
+                elif isinstance(value, str):
+                    if value.startswith("<function "):
+                        continue
+                    elif var_name.endswith("_id"):
+                        continue
+                frame["vars"][var_name] = "<redacted>"
+    return event
+
+
+@defaults(
+    enabled=False,
+    dsn=None,
```

Comment:
It took me a while to figure out what `dsn` stands for (very sentry specific), maybe a comment? Or accept full `data_source_name` in config?

Reply:
Good point will add a comment, as sorta locked into the config naming due to using this in the pubsub implementation. Will look into updating these when these get replaced by a lib (also probably worth adding a `_url` suffix too)